tests/meta/workflow/mutant/test_mutant_quality_controller.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+from cg.meta.workflow.mutant.quality_controller.models import (
+    MutantPoolSamples,
+    QualityResult,
+    CaseQualityResult,
+    QualityMetrics,
+    SampleQualityResults,
+    SampleResults,
+    SamplesQualityResults,
+)
+from cg.meta.workflow.mutant.quality_controller.quality_controller import (
+    MutantQualityController,
+)
+from cg.store.models import Case, Sample
+
+
+def test__get_mutant_pool_samples(
+    mutant_quality_controller: MutantQualityController,
+    mutant_case_qc_pass: Case,
+    sample_qc_pass: Sample,
+    external_negative_control_qc_pass: Sample,
+    internal_negative_control_qc_pass: Sample,
+):
+    # WHEN creating a MutantPoolSamples object
+    mutant_pool_samples: MutantPoolSamples = mutant_quality_controller._get_mutant_pool_samples(
+        case=mutant_case_qc_pass
+    )
+
+    # THEN the pool is created correctly:
+    #   - the external negative control is identified and separated from the rest of the samples
+    #   - all other samples are present in the list under samples
+    #   - the internal negative control corresponding to the case is fetched from lims and added to the pool
+
+    assert mutant_pool_samples.external_negative_control == external_negative_control_qc_pass
+    assert mutant_pool_samples.samples == [sample_qc_pass]
+    assert mutant_pool_samples.internal_negative_control == internal_negative_control_qc_pass
+
+
+def test_get_quality_metrics(
+    mutant_quality_controller: MutantQualityController,
+    mutant_results_file_path_case_qc_pass: Path,
+    mutant_case_qc_pass: Case,
+    mutant_sample_results_sample_qc_pass: SampleResults,
+    sample_qc_pass: Sample,
+):
+    # GIVEN a case
+
+    # WHEN generating the quality_metrics
+    quality_metrics: QualityMetrics = mutant_quality_controller._get_quality_metrics(
+        case_results_file_path=mutant_results_file_path_case_qc_pass,
+        case=mutant_case_qc_pass,
+    )
+
+    # THEN no errors are raised and the sample_results are created for each sample
+    assert (
+        quality_metrics.results[sample_qc_pass.internal_id] == mutant_sample_results_sample_qc_pass
+    )
+
+
+def test__get_sample_quality_result_for_sample(
+    mutant_quality_controller: MutantQualityController,
+    sample_qc_pass: Sample,
+    mutant_sample_results_sample_qc_pass: SampleResults,
+):
+    # GIVEN a sample that passes qc and its corresponding SampleResults
+
+    # WHEN peforming quality control on the sample
+    sample_quality_results_sample_qc_pass: SampleQualityResults = (
+        mutant_quality_controller._get_sample_quality_result_for_sample(
+            sample=sample_qc_pass,
+            sample_results=mutant_sample_results_sample_qc_pass,
+        )
+    )
+    # THEN the sample passes qc
+    assert sample_quality_results_sample_qc_pass.passes_qc is True
+
+
+def test__get_sample_quality_result_for_internal_negative_control_sample(
```

Comment:
```suggestion
def test_get_sample_quality_result_for_internal_negative_control_sample(
```

Reply:
hahaha there are more but you got the idea, I'm spamming at this point 😆 🙏

Reply:
Will change all of them! 😁